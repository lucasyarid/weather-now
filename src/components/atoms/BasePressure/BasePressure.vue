<template>
  <div class="BasePressure">
    <div class="BasePressure__title">
      Pressure
    </div>
    <div class="BasePressure__value">
      {{ pressure }}<span class="BasePressure__scale">hPa</span>
    </div>
  </div>
</template>

<script>
export default {
  name: "BasePressure",
  props: {
    pressure: {
      type: Number,
      required: true
    }
  }
};
</script>

<style lang="scss" scoped>
.BasePressure {
  color: $color-darker;
  text-align: center;
  &__title {
    text-transform: uppercase;
    font-size: 12px;
    margin-bottom: 5px;
    font-weight: 100;
  }
  &__value {
    font-size: 21px;
  }
  &__scale {
    font-size: 12px;
  }
}
</style>
