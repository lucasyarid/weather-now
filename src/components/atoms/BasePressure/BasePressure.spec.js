import { shallowMount } from "@vue/test-utils";
import BasePressure from "./BasePressure";

describe("BasePressure", () => {
  const WRAPPER = shallowMount(BasePressure, {
    propsData: {
      pressure: 892
    }
  });

  test("Componente existe", () => {
    expect(WRAPPER.exists()).toBe(true);
  });

  test("Componente nÃ£o alterado", () => {
    expect(WRAPPER.html()).toMatchSnapshot();
  });
});
