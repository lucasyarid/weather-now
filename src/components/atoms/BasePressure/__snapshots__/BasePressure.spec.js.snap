// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasePressure Componente n√£o alterado 1`] = `
"<div class=\\"BasePressure\\"><div class=\\"BasePressure__title\\">
    Pressure
  </div> <div class=\\"BasePressure__value\\">
    892<span class=\\"BasePressure__scale\\">hPa</span></div></div>"
`;
